app/containers/StatusBar.tsx

```diff
@@ -15,14 +14,20 @@ interface IStatusBar {
 }
 
 const StatusBar = React.memo(({ barStyle, backgroundColor }: IStatusBar) => {
-	const { theme } = useTheme();
-	if (!barStyle) {
-		barStyle = 'light-content';
-		if (theme === 'light') {
-			barStyle = 'dark-content';
+	const { theme, colors } = useTheme();
+
+	useEffect(() => {
+		if (!barStyle) {
+			barStyle = 'light-content';
+			if (theme === 'light') {
+				barStyle = 'dark-content';
+			}
 		}
-	}
-	return <StatusBarRN backgroundColor={backgroundColor ?? themes[theme].headerBackground} barStyle={barStyle} animated />;
+		StatusBarRN.setBackgroundColor(backgroundColor ?? colors.headerBackground);
```

Comment:
Are these methods only for android?
Why do you set them here without doing a platform check, but the fix is for android.
Have you checked to see if anything happens on iOS?

Reply:
the method setBackgroundColor is only for Android, the same as passing the backgroundColor by props. Works the same as props, and doesn't affect the iOS. As before, we don't need to check the platform. wdyt?

Reply:
these method**S**
The method marked above is only for android, but the one below is not.

Reply:
Sure, the below`StatusBarRN.setBarStyle` works for both and is the same as passing barStyle as props, also fix the Android bug. 

I decided to do on this way just to keep the same pattern for both, instead of separating the StatusBar component for each platform